### ISSUE TITLE
GKE AutoPilot - No Write Mode Hostpath

### ISSUE THREAD

**Issue by brianjflowhub** [elastic/helm-charts#1126]
Tagged version: 7.12.0

Is it possible to run APM-Server on GKE AutoPilot where GKE best security practice are required? Here's the error I'm receiving when I deploy the helm chart.

```
Error from server ([denied by autogke-no-write-mode-hostpath] hostPath volume data in container busybox is accessed in write mode; disallowed in Autopilot. Requesting user: <gitlab-deploy@company-dev.iam.gserviceaccount.com> and groups: <["system:authenticated"]>
[denied by autogke-no-write-mode-hostpath] hostPath volume data in container apm-server is accessed in write mode; disallowed in Autopilot. Requesting user: <gitlab-deploy@company-dev.iam.gserviceaccount.com> and groups: <["system:authenticated"]>): error when creating "1122201989/manifest.yaml": admission webhook "validation.gatekeeper.sh" denied the request: [denied by autogke-no-write-mode-hostpath] hostPath volume data in container busybox is accessed in write mode; disallowed in Autopilot. Requesting user: <gitlab-deploy@company-dev.iam.gserviceaccount.com> and groups: <["system:authenticated"]>
[denied by autogke-no-write-mode-hostpath] hostPath volume data in container apm-server is accessed in write mode; disallowed in Autopilot. Requesting user: <gitlab-deploy@company-dev.iam.gserviceaccount.com> and groups: <["system:authenticated"]>
```

**Comment by jmlrt** [comment]
Hi @brianjflowhub,
We didn't investigate into GKE AutoPilot support yet.
Without knowing anything about it, I see references to hostpath volume and busybox container, however this chart doesn't use any busybox image and the only mounted volume is the configMap with default values.

**Comment by joshpurvis** [comment]
I ran into a similar error on the ES chart. 

```
Error: failed to create resource: admission webhook "validation.gatekeeper.sh" denied the request: [denied by autogke-disallow-privilege] container <configure-sysctl> is privileged; not allowed in Autopilot. Requesting user: <REDACTED@REDACTED.iam.gserviceaccount.com> and groups: <["system:authenticated"]>
```

After some digging, I came across an issue in an unrelated project: https://github.com/DataDog/helm-charts/issues/182

According to one of the comments there, their chart works in GKE AutoPilot's "rapid release" channel -- so I assume the issue is GCP's problem and will eventually resolve itself in the next release?

**Comment by jmlrt** [comment]
> I ran into a similar error on the ES chart.
> 
> ```
> Error: failed to create resource: admission webhook "validation.gatekeeper.sh" denied the request: [denied by autogke-disallow-privilege] container <configure-sysctl> is privileged; not allowed in Autopilot. Requesting user: <REDACTED@REDACTED.iam.gserviceaccount.com> and groups: <["system:authenticated"]>
> ```
> 
> After some digging, I came across an issue in an unrelated project: [DataDog/helm-charts#182](https://github.com/DataDog/helm-charts/issues/182)
> 
> According to one of the comments there, their chart works in GKE AutoPilot's "rapid release" channel -- so I assume the issue is GCP's problem and will eventually resolve itself in the next release?

This error is different and is related to the initContainer running on privileged mode to configure virtual memory on K8S nodes (https://www.elastic.co/guide/en/elasticsearch/reference/current/vm-max-map-count.html#vm-max-map-count).

If AutoPilot don't allow privileged container, you can still disable it using `sysctlInitContainer.enabled=false`

**Comment by alphayax** [comment]
> > I ran into a similar error on the ES chart.
> > ```
> > Error: failed to create resource: admission webhook "validation.gatekeeper.sh" denied the request: [denied by autogke-disallow-privilege] container <configure-sysctl> is privileged; not allowed in Autopilot. Requesting user: <REDACTED@REDACTED.iam.gserviceaccount.com> and groups: <["system:authenticated"]>
> > ```
> > 
> > 
> > After some digging, I came across an issue in an unrelated project: [DataDog/helm-charts#182](https://github.com/DataDog/helm-charts/issues/182)
> > According to one of the comments there, their chart works in GKE AutoPilot's "rapid release" channel -- so I assume the issue is GCP's problem and will eventually resolve itself in the next release?
> 
> This error is different and is related to the initContainer running on privileged mode to configure virtual memory on K8S nodes (https://www.elastic.co/guide/en/elasticsearch/reference/current/vm-max-map-count.html#vm-max-map-count).
> 
> If AutoPilot don't allow privileged container, you can still disable it using `sysctlInitContainer.enabled=false`

I've got the same error on the ES Chart too.
But setting `sysctlInitContainer.enabled=false` doesn't help : ES doesn't start because memory is too low without the sysctl init container.

```
ERROR: [1] bootstrap checks failed
[1]: max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]
```

**Comment by maiconbaum** [comment]
> > > I ran into a similar error on the ES chart.
> > > ```
> > > Error: failed to create resource: admission webhook "validation.gatekeeper.sh" denied the request: [denied by autogke-disallow-privilege] container <configure-sysctl> is privileged; not allowed in Autopilot. Requesting user: <REDACTED@REDACTED.iam.gserviceaccount.com> and groups: <["system:authenticated"]>
> > > ```
> > > 
> > > 
> > > After some digging, I came across an issue in an unrelated project: [DataDog/helm-charts#182](https://github.com/DataDog/helm-charts/issues/182)
> > > According to one of the comments there, their chart works in GKE AutoPilot's "rapid release" channel -- so I assume the issue is GCP's problem and will eventually resolve itself in the next release?
> > 
> > 
> > This error is different and is related to the initContainer running on privileged mode to configure virtual memory on K8S nodes (https://www.elastic.co/guide/en/elasticsearch/reference/current/vm-max-map-count.html#vm-max-map-count).
> > If AutoPilot don't allow privileged container, you can still disable it using `sysctlInitContainer.enabled=false`
> 
> I've got the same error on the ES Chart too.
> But setting `sysctlInitContainer.enabled=false` doesn't help : ES doesn't start because memory is too low without the sysctl init container.
> 
> ```
> ERROR: [1] bootstrap checks failed
> [1]: max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]
> ```

This is still an issue.

**Comment by orlandothoeny** [comment]
This is still an ongoing issue. Blocks running Elasticserarch on a GKE Autopilot cluster.

**Comment by jvkubjg** [comment]
Any updates on that? 
We want to run ES on GKE Autopilot

**Comment by nakabonne** [comment]
I'm running into the `sysctlInitContainer` issue too. Forcing us to run containers in privileged mode is not desirable for everyone from a security standpoint, not just for GKE Autopilot.

Hopefully we have another way to increase the operating system limits on `mmap` counts.

**Comment by omrta-dev** [comment]
Also running into the same issue and there doesn't seem to be a way to solve this yet. GKE Autopilot prevents privileged containers and there doesn't seem to be any way to increase mmap outside of that. I honestly don't know if there is much the elastic team can do to solve this issue but i'd love to see it solved by GKE.

**Comment by keeth** [comment]
Wonder if this is on their roadmap or if we can somehow get the attention of the GKE team 🤔

**Comment by nhat2008** [comment]
you can add 2 following params to the config 
      node.store.allow_mmap: false
      index.store.type: niofs 

ES - 7.9.2

**Comment by bat9r** [comment]
Also stacked with this error

**Comment by viniciusesteter** [comment]
Also stacked with this error [2] @elasticitservice

**Comment by stefluhh** [comment]
Disabling mmap is discouraged by ES: [Click](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-virtual-memory.html) For process critical use cases I would simply not run ES on Autopilot. Is understandable that Autopilot doesn't allow changing kernel settings by design of GKE AP.

**Comment by keeth** [comment]
@stefano-lucka yes we have given up on Autopilot and are now using cloud.elastic.co

**Comment by Prashanth073-cloud** [comment]
I am also facing the same issue. ES is giving issues on GKE Auto pilot. Does anyone fix this issue on GKE Auto pilot? Is there any alternative?

**Comment by vasi-grigo** [comment]
up

**Comment by dorkamotorka** [comment]
+1
